ko/Tiltfile

```diff
@@ -0,0 +1,30 @@
+# -*- mode: Python -*-
+
+def ko_build(ref, import_path, deps=[]):
+  """Use Ko (https://github.com/google/ko) to build images for Tilt.
+
+  Args:
+    ref: The name of the image to build. Must match the image
+      name in the Kubernetes resources you're deploying.
+    import_path: A Go import path of the binary to build.
+    deps: A list of dependencies that can trigger rebuilds.
+  """
+  commands = [
+    "set -eo pipefail",
+  ]
+
+  if not import_path.startswith("."):
+    # Ko needs the source code to be checked out.
+    # https://github.com/google/ko/issues/237
+    commands += ["go get {import_path}".format(import_path=import_path)]
+
+  commands += [
+    "IMAGE=$(ko publish --local {import_path})".format(import_path=import_path),
+    'if [[ "$?" != "0" ]]; then exit 1; fi',
+    "docker tag $IMAGE $EXPECTED_REF",
+  ]
+  custom_build(
+    ref=ref,
+    command=["bash", "-c", ";\n".join(commands)],
+    deps=deps + [__file__],
```

Comment:
Just to confirm my understanding, we only need the `__file__` here to catch iterations on the extension itself that don't affect any of the custom_build args, right?

Reply:
no, this is a workaround for https://github.com/tilt-dev/tilt/issues/3891, left a comment about this